clusterloader2/docs/experiments.md

```diff
@@ -0,0 +1,44 @@
+
+For this reason, all new changes features added to clusterloader2, shall be gated and rolled out gradually. This way we can minimize blast radius of breaking changes. We can even remove some kinds of them to happen at all, as they should be caught at presubmit time and not allowed to merge at all.
+
+## Experiment rollout
+
+### General principles
+
+1. If for some reason it doesn't make sense to enable an experiment in particular test job indicate the reason clearly in the PR (e.g. measuring kube-proxy metrics in kubemark can be skipped as kube-proxy is not running there).
+
+1. Allow at least 24h between previous step to ensure the experiment is stable. We don't want to block PR to k/k because of a flaky experiment in clusterloader2.
+
+1. Check with sig-scalability whether there is a regression in the test you want to enable experiment for. We don't want new features in clusterloader2 to interfere with regression' debugging.
+
+1. For tests not listed below it's fine to enable experiments
+
+### Step-by-step process
+
+_Each step should be a separate PR_
+
+1. Enable experiment in [perf-tests presubmits](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml#L267) and [ci-kubernetes-e2e-gci-gce-scalability](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-release-blocking-jobs.yaml#L98)
+
+   Perf-test presubmit runs two jobs: [pull-perf-tests-clusterloader2](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml#L268) and [pull-perf-tests-clusterloader2-kubemark](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml#L317). Primary role of those presubmits is to catch bugs in code from perf-tests, so enable any experiment for both of those jobs first. Before enabling the experiment in [ci-kubernetes-e2e-gci-gce-scalability](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-release-blocking-jobs.yaml#L98) make sure there is no ongoing regression affecting this test.
+
+1. Enable experiment in [pull-kubernetes-e2e-gce-100-performance](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml#L3) and [pull-kubernetes-kubemark-e2e-gce-big](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml#L149)
+
+   Block changes to k/k if they don't work with the experiment enabled.
```

Comment:
Could you rewrite this to make it clear what is the intention? 

I think there should be some additional procedure here, this step is the most prone to blocking k/k changes, due to the fact that you modify the presubmit via PR in test-infra, which doesn't trigger the k/k presubmit to run. It's very easy to  break the k/k presubmit if you don't keep an eye on k/k presubmits directly after your test-infra PR has merged

Reply:
Added more details. WDYT?